### PR TITLE
Add Travis

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/navigation/issues/25

Add Travis integration to run tests/linting on pull requests to `main`.

### CODE REVIEW

**Comment by psealock** [comment]
The repo doesn't show up in https://travis-ci.com/organizations/woocommerce/repositories

I'm using this for reference https://docs.travis-ci.com/user/tutorial/

**Comment by psealock** [comment]
Permissions have been added https://travis-ci.com/github/woocommerce/navigation.

Unfortunately, I think this may require `.travis.yml` to be in the main branch to kick off builds for pull requests, which makes testing this kind of hard.

**Comment by psealock** [comment]
Any objection to merging this and dealing with issues that come up separately? The local setup [seems a bit involved](https://gist.github.com/fulldecent/d84cd1abdcc6930865d1b862c4aed917).

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - bash bin/install-wp-tests.sh wc_admin_test root '' localhost $WP_VERSION
+  - bash bin/travis.sh before
+  - node --version
+  - npm --version
+  - timedatectl
+
+script:
+  - bash bin/js_lint_test.sh
+  - bash bin/phpunit.sh
+  - bash bin/phpcs.sh
```
Comment (jeffstieler) [inline .travis.yml:44]: I don't see this file in the tree.

File: bin/js_lint_test.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#!/usr/bin/env bash
+
+set -o errexit
+
+if [[ ${RUN_JS} == 1 ]]; then
+	npm run -s install-if-deps-outdated
+	npm run lint:js client/**/*.js
+	npm run build
+	npm run test:unit
+fi
```
Comment (jeffstieler) [inline bin/js_lint_test.sh:11]: No newline at end of file

File: bin/travis.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+# usage: travis.sh before|after
+
+if [ "$1" == 'before' ]; then
+	if [[ "$COMPOSER_DEV" == "1" ]]; then
+		composer install
+	else
+		composer install --no-dev
+	fi
+fi
```
Comment (jeffstieler) [inline bin/travis.sh:10]: No newline at end of file

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+before_install:
+  - nvm install 'lts/*'
+
+before_script:
+  - phpenv config-rm xdebug.ini
+  - export PATH="$WP_CORE_DIR/wp-content/plugins/navigation/vendor/bin:$PATH"
+  - bash bin/install-wp-tests.sh wc_admin_test root '' localhost $WP_VERSION
+  - bash bin/travis.sh before
```
Comment (jeffstieler) [inline .travis.yml:36]: This file doesn't appear to be in the tree.

**Comment by psealock** [comment]
PHP unit tests aren't running. I decided to merge as is so we have something and pull out the PHP issue to a follow up issue in https://github.com/woocommerce/navigation/issues/55.

cc @joshuatf

**Pull request merged**